In [37]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn import *
%matplotlib inline

In [38]:
heartAtk = pd.read_csv("./heart_attack_prediction_dataset.csv")
heartAtk.columns

Index(['Patient ID', 'Age', 'Sex', 'Cholesterol', 'Blood Pressure',
       'Heart Rate', 'Diabetes', 'Family History', 'Smoking', 'Obesity',
       'Alcohol Consumption', 'Exercise Hours Per Week', 'Diet',
       'Previous Heart Problems', 'Medication Use', 'Stress Level',
       'Sedentary Hours Per Day', 'Income', 'BMI', 'Triglycerides',
       'Physical Activity Days Per Week', 'Sleep Hours Per Day', 'Country',
       'Continent', 'Hemisphere', 'Heart Attack Risk'],
      dtype='object')

In [39]:
heartAtk.shape

(8763, 26)

In [40]:
heartAtk.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8763 entries, 0 to 8762
Data columns (total 26 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Patient ID                       8763 non-null   object 
 1   Age                              8763 non-null   int64  
 2   Sex                              8763 non-null   object 
 3   Cholesterol                      8763 non-null   int64  
 4   Blood Pressure                   8763 non-null   object 
 5   Heart Rate                       8763 non-null   int64  
 6   Diabetes                         8763 non-null   int64  
 7   Family History                   8763 non-null   int64  
 8   Smoking                          8763 non-null   int64  
 9   Obesity                          8763 non-null   int64  
 10  Alcohol Consumption              8763 non-null   int64  
 11  Exercise Hours Per Week          8763 non-null   float64
 12  Diet                

In [41]:
heartAtk.head()

Patient ID  Age     Sex  Cholesterol Blood Pressure  Heart Rate  Diabetes  \
0    BMW7812   67    Male          208         158/88          72         0   
1    CZE1114   21    Male          389         165/93          98         1   
2    BNI9906   21  Female          324         174/99          72         1   
3    JLN3497   84    Male          383        163/100          73         1   
4    GFO8847   66    Male          318          91/88          93         1   

   Family History  Smoking  Obesity  ...  Sedentary Hours Per Day  Income  \
0               0        1        0  ...                 6.615001  261404   
1               1        1        1  ...                 4.963459  285768   
2               0        0        0  ...                 9.463426  235282   
3               1        1        0  ...                 7.648981  125640   
4               1        1        1  ...                 1.514821  160555   

         BMI  Triglycerides  Physical Activity Days Per Week  \
0  31.251233            286                                0   
1  27.194973            235                                1   
2  28.176571            587                                4   
3  36.464704            378                                3   
4  21.809144            231                                1   

   Sleep Hours Per Day    Country      Continent           Hemisphere  \
0                    6  Argentina  South America  Southern Hemisphere   
1                    7     Canada  North America  Northern Hemisphere   
2                    4     France         Europe  Northern Hemisphere   
3                    4     Canada  North America  Northern Hemisphere   
4                    5   Thailand           Asia  Northern Hemisphere   

   Heart Attack Risk  
0                  0  
1                  0  
2                  0  
3                  0  
4                  0  

[5 rows x 26 columns]

In [42]:
columns_to_drop=['Patient ID','Hemisphere','Continent']
heartAtk_clean = heartAtk.drop(columns_to_drop,axis=1)

In [43]:
X = heartAtk_clean.drop(['Heart Attack Risk'],axis=1)
y = heartAtk_clean['Heart Attack Risk'].to_frame()
y

Heart Attack Risk
0                     0
1                     0
2                     0
3                     0
4                     0
...                 ...
8758                  0
8759                  0
8760                  1
8761                  0
8762                  1

[8763 rows x 1 columns]

In [44]:
heartAtk_clean.shape

(8763, 23)

In [45]:
heartAtk_clean.describe()

Age  Cholesterol   Heart Rate     Diabetes  Family History  \
count  8763.000000  8763.000000  8763.000000  8763.000000     8763.000000   
mean     53.707977   259.877211    75.021682     0.652288        0.492982   
std      21.249509    80.863276    20.550948     0.476271        0.499979   
min      18.000000   120.000000    40.000000     0.000000        0.000000   
25%      35.000000   192.000000    57.000000     0.000000        0.000000   
50%      54.000000   259.000000    75.000000     1.000000        0.000000   
75%      72.000000   330.000000    93.000000     1.000000        1.000000   
max      90.000000   400.000000   110.000000     1.000000        1.000000   

           Smoking      Obesity  Alcohol Consumption  Exercise Hours Per Week  \
count  8763.000000  8763.000000          8763.000000              8763.000000   
mean      0.896839     0.501426             0.598083                10.014284   
std       0.304186     0.500026             0.490313                 5.783745   
min       0.000000     0.000000             0.000000                 0.002442   
25%       1.000000     0.000000             0.000000                 4.981579   
50%       1.000000     1.000000             1.000000                10.069559   
75%       1.000000     1.000000             1.000000                15.050018   
max       1.000000     1.000000             1.000000                19.998709   

       Previous Heart Problems  Medication Use  Stress Level  \
count              8763.000000     8763.000000   8763.000000   
mean                  0.495835        0.498345      5.469702   
std                   0.500011        0.500026      2.859622   
min                   0.000000        0.000000      1.000000   
25%                   0.000000        0.000000      3.000000   
50%                   0.000000        0.000000      5.000000   
75%                   1.000000        1.000000      8.000000   
max                   1.000000        1.000000     10.000000   

       Sedentary Hours Per Day         Income          BMI  Triglycerides  \
count              8763.000000    8763.000000  8763.000000    8763.000000   
mean                  5.993690  158263.181901    28.891446     417.677051   
std                   3.466359   80575.190806     6.319181     223.748137   
min                   0.001263   20062.000000    18.002337      30.000000   
25%                   2.998794   88310.000000    23.422985     225.500000   
50%                   5.933622  157866.000000    28.768999     417.000000   
75%                   9.019124  227749.000000    34.324594     612.000000   
max                  11.999313  299954.000000    39.997211     800.000000   

       Physical Activity Days Per Week  Sleep Hours Per Day  Heart Attack Risk  
count                      8763.000000          8763.000000        8763.000000  
mean                          3.489672             7.023508           0.358211  
std                           2.282687             1.988473           0.479502  
min                           0.000000             4.000000           0.000000  
25%                           2.000000             5.000000           0.000000  
50%                           3.000000             7.000000           0.000000  
75%                           5.000000             9.000000           1.000000  
max                           7.000000            10.000000           1.000000

In [46]:
heartAtk_clean.isna().any()

Age                                False
Sex                                False
Cholesterol                        False
Blood Pressure                     False
Heart Rate                         False
Diabetes                           False
Family History                     False
Smoking                            False
Obesity                            False
Alcohol Consumption                False
Exercise Hours Per Week            False
Diet                               False
Previous Heart Problems            False
Medication Use                     False
Stress Level                       False
Sedentary Hours Per Day            False
Income                             False
BMI                                False
Triglycerides                      False
Physical Activity Days Per Week    False
Sleep Hours Per Day                False
Country                            False
Heart Attack Risk                  False
dtype: bool

In [47]:
heartAtk_clean.duplicated().sum()
heartAtk_clean.drop_duplicates(inplace=True)
heartAtk_clean.head()

Age     Sex  Cholesterol Blood Pressure  Heart Rate  Diabetes  \
0   67    Male          208         158/88          72         0   
1   21    Male          389         165/93          98         1   
2   21  Female          324         174/99          72         1   
3   84    Male          383        163/100          73         1   
4   66    Male          318          91/88          93         1   

   Family History  Smoking  Obesity  Alcohol Consumption  ...  Medication Use  \
0               0        1        0                    0  ...               0   
1               1        1        1                    1  ...               0   
2               0        0        0                    0  ...               1   
3               1        1        0                    1  ...               0   
4               1        1        1                    0  ...               0   

  Stress Level  Sedentary Hours Per Day  Income        BMI  Triglycerides  \
0            9                 6.615001  261404  31.251233            286   
1            1                 4.963459  285768  27.194973            235   
2            9                 9.463426  235282  28.176571            587   
3            9                 7.648981  125640  36.464704            378   
4            6                 1.514821  160555  21.809144            231   

   Physical Activity Days Per Week  Sleep Hours Per Day    Country  \
0                                0                    6  Argentina   
1                                1                    7     Canada   
2                                4                    4     France   
3                                3                    4     Canada   
4                                1                    5   Thailand   

   Heart Attack Risk  
0                  0  
1                  0  
2                  0  
3                  0  
4                  0  

[5 rows x 23 columns]

In [52]:
def string_int_systolic(string):
    parts = string.split('/')
    return int(parts[0])

def string_int_diastolic(string):
    parts = string.split('/')
    return int(parts[1])


df = pd.DataFrame(heartAtk_clean)
df['Systolic Pressure'] = df['Blood Pressure'].apply(string_int_systolic)
df['Diastolic Pressure'] = df['Blood Pressure'].apply(string_int_diastolic)


Blood Pressure Values by category
Category/Systolic/Diastolic
Normal : less than 120 and less than 80

Elevated : 120-129 amd less than 80

High Blood Pressure (Hypertension) Stage 1 : 130-139 or 80-89

High Blood Pressure (Hypertension) Stage 2 : 140 or higher or 90 or higher

Hypertensive Crisis : higher than 180 and/or higher than 120

Relevant age groups: 18-39 ; 40-59; 60+

In [69]:
labels = ['18-39', '40-59', '60+']
bins = [18, 39, 59, 110]

# Create 'Age Group' column
df['Age Group'] = pd.cut(df['Age'], bins=bins, labels=labels, right=False)

# Group by 'Age Group' and 'Sex'
grouped = df.groupby(['Age Group', 'Sex'])

# Get specific groups (e.g., males and females in the 18-39 age group)
age_group_1839_men = grouped.get_group(('18-39', 'Male'))
age_group_1839_women = grouped.get_group(('18-39', 'Female'))

# Similarly, you can retrieve other age groups and sex-specific groups
age_group_4059_men = grouped.get_group(('40-59', 'Male'))
age_group_4059_women = grouped.get_group(('40-59', 'Female'))

age_group_60over_men = grouped.get_group(('60+', 'Male'))
age_group_60over_women = grouped.get_group(('60+', 'Female'))


age_group_1839_men.head()

/tmp/ipykernel_4363/1563590602.py:8: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  grouped = df.groupby(['Age Group', 'Sex'])


Age   Sex  Cholesterol Blood Pressure  Heart Rate  Diabetes  \
1    21  Male          389         165/93          98         1   
8    20  Male          145        144/105          68         1   
17   38  Male          166         120/74          56         1   
24   36  Male          133         161/90          97         1   
31   25  Male          197         178/72          45         0   

    Family History  Smoking  Obesity  Alcohol Consumption  ...  Income  \
1                1        1        1                    1  ...  285768   
8                0        1        1                    0  ...   25086   
17               0        1        1                    0  ...   48376   
24               0        1        1                    1  ...  223132   
31               1        1        0                    1  ...   59634   

          BMI  Triglycerides  Physical Activity Days Per Week  \
1   27.194973            235                                1   
8   35.809901            790                                7   
17  29.517388            402                                0   
24  22.386804            605                                5   
31  18.522199            661                                0   

    Sleep Hours Per Day   Country  Heart Attack Risk  Systolic Pressure  \
1                     7    Canada                  0                165   
8                     4    Brazil                  0                144   
17                    6  Thailand                  0                120   
24                   10    Canada                  0                161   
31                    6     Spain                  0                178   

    Diastolic Pressure  Age Group  
1                   93      18-39  
8                  105      18-39  
17                  74      18-39  
24                  90      18-39  
31                  72      18-39  

[5 rows x 26 columns]

In [72]:
mean_age39_systolic_men = age_group_1839_men['Systolic Pressure'].mean()
mean_age39_diastolic_men = age_group_1839_men['Diastolic Pressure'].mean()

mean_age39_systolic_women = age_group_1839_women['Systolic Pressure'].mean()
mean_age39_diastolic_women = age_group_1839_women['Diastolic Pressure'].mean()

mean_age59_systolic_men = age_group_4059_men['Systolic Pressure'].mean()
mean_age59_diastolic_men = age_group_4059_men['Diastolic Pressure'].mean()

mean_age59_systolic_women = age_group_4059_women['Systolic Pressure'].mean()
mean_age59_diastolic_women = age_group_4059_women['Diastolic Pressure'].mean()

mean_age60over_systolic_men = age_group_60over_men['Systolic Pressure'].mean()
mean_age60over_diastolic_men = age_group_60over_men['Diastolic Pressure'].mean()

mean_age60over_systolic_women = age_group_60over_women['Systolic Pressure'].mean()
mean_age60over_diastolic_women = age_group_60over_women['Diastolic Pressure'].mean()


print("Blood pressure means on men (age group 18-39) :\n Systolic %f Diastolic %f" %(mean_age39_systolic_men, mean_age39_diastolic_men) )
print("Blood pressure means on women (age group 18-39) :\n Systolic %f Diastolic %f\n" %(mean_age39_systolic_women, mean_age39_diastolic_women) )

print("Blood pressure means on men (age group 40-59) :\n Systolic %f Diastolic %f" %(mean_age59_systolic_men, mean_age59_diastolic_men) )
print("Blood pressure means on women (age group 40-59) :\n Systolic %f Diastolic %f\n" %(mean_age59_systolic_women, mean_age59_diastolic_women) )

print("Blood pressure means on men (age group 60+) :\n Systolic %f Diastolic %f" %(mean_age60over_systolic_men, mean_age60over_diastolic_men) )
print("Blood pressure means on women (age group 60+) :\n Systolic %f Diastolic %f\n" %(mean_age60over_systolic_women, mean_age60over_diastolic_women) )


Blood pressure means on men (age group 18-39) :
 Systolic 134.558413 Diastolic 84.931247
Blood pressure means on women (age group 18-39) :
 Systolic 135.510740 Diastolic 85.658711

Blood pressure means on men (age group 40-59) :
 Systolic 134.773538 Diastolic 85.468308
Blood pressure means on women (age group 40-59) :
 Systolic 135.515328 Diastolic 85.979562

Blood pressure means on men (age group 60+) :
 Systolic 135.363367 Diastolic 85.067853
Blood pressure means on women (age group 60+) :
 Systolic 135.053144 Diastolic 84.401240



Só no grupo de 60+ é que os valores se encontram dentro da média